demos/999-storage-import-vhd-blob/azuredeploy.json

```diff
@@ -0,0 +1,169 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.210508",
+    "parameters": {
+        "location": {
+            "type": "string",
+            "defaultValue": "[resourceGroup().location]",
+            "metadata": {
+                "description": "The location in which the resources should be deployed."
+            }
+        },
+        "Source": {
+            "type": "string",
+            "metadata": {
+                "description": "URI of the VHD to fetch (Blob URI with SAS Token). For example: https://source-blob-storage-name.blob.core .windows.net/source-container-name?sv=1234&spr=https&sig=1234"
+            }
+        },
+        "Destination": {
+            "type": "string",
+            "metadata": {
+                "description": "URI of the Storage Account container with a writable SAS Token. For example: https://blob-storage.blob.core .windows.net/my-container-name?sv=123412341234&spr=https&sig=123412341234"
+            }
+        }
+    },
+    "functions": [],
+    "variables": {
+        "uriVHDBlobSasToken": "[parameters('Source')]",
+        "uriWritableStorageAccountBlobContainerSasToken": "[parameters('Destination')]",
+        "nestedTemplateName_ImportVHD-FromBlobUri": "[concat('ImportVHD-FromBlobUri-',uniqueString(parameters('Source'),uniqueString(parameters('Destination'))))]"
+    },
+    "resources": [
+        {
+            "type": "Microsoft.Resources/deployments",
```

Comment:
why is this nested?

Reply:
It is mainly for the "Deploy in Azure" button experience. It is simpler for users who may try the template, there is less to fill and understand when trying it. 

The nesting will hide the "technical" parameters in the form that shows up when deploying via the Azure Portal and at the same time the nested template keeps the parameters/structure we need when integrating into another template. 

Initially i wanted to hide the "location" parameter too, so that user would just have to fill the Region at the resource group level ( (and then store the resource group region in variable) but i think it is not allowed with best practices. I have couple of other sample like that for demos like that one: https://github.com/riverbed/Riverbed-Community-Toolkit/tree/master/AppResponse/Azure-Cloud-Cookbooks/101-appresponse-simple-demo

Reply:
Removed nesting done for storage-import-zipped-vhds (https://github.com/Azure/azure-quickstart-templates/pull/9561)